src/timely-util/src/operator.rs

```diff
@@ -708,7 +707,18 @@ where
                 h.finish()
             });
             consolidate_pact::<Ba, _, _>(&self.map(|k| (k, ())).inner, exchange, name)
-                .map(|((k, ()), time, diff)| (k.clone(), time.clone(), diff.clone()))
+                .unary(Pipeline, "unpack consolidated", |_, _| {
```

Comment:
Nit, but if this is coming in as a `Rc<Vec<...>>` we could check for unique ownership and avoid cloning everything, I think?

Reply:
It's columnar data, so we can only get references out of it!

Reply:
I see! Owned containers, but the containers don't own keys, times, etc.